### PR TITLE
Support SHA256 fingerprints for host key validation

### PR DESCRIPTION
**Pull request by DavisNT**
This PR adds support for host key validation using SHA256 hash in Base64 (with padding removed) format commonly used by other SSH software (OpenSSH, PuTTY etc.).

### CODE REVIEW

**Comment by DavisNT** [comment]
I force-pushed the same code several times (`git commit --amend` and `git push -f`) to re-run the failed AppVeyor build (tests apparently unrelated to these changes were failing - details [here](https://ci.appveyor.com/project/drieseng/ssh-net/builds/46591305/tests), [here](https://ci.appveyor.com/project/drieseng/ssh-net/builds/46591384/tests) and [here](https://ci.appveyor.com/project/drieseng/ssh-net/builds/46600867/tests)).

**Comment by DavisNT** [comment]
@drieseng Can you please look into this change?

File: src/Renci.SshNet/Common/HostKeyEventArgs.cs [hunk 1]
```diff
@@ -58,6 +69,11 @@ public HostKeyEventArgs(KeyHostAlgorithm host)
             {
                 FingerPrint = md5.ComputeHash(host.Data);
             }
+
+            using (var sha256 = CryptoAbstraction.CreateSHA256())
+            {
+                FingerPrintSHA256 = Convert.ToBase64String(sha256.ComputeHash(host.Data)).Replace("=", "");
```
Comment (WojciechNagorski) [inline src/Renci.SshNet/Common/HostKeyEventArgs.cs:75]: Why here is `.Replace("=", "")`? Do other programs work like that?

Comment (DavisNT) [reply src/Renci.SshNet/Common/HostKeyEventArgs.cs:75]: Yes, other software works like this as well.

OpenSSH does this.
For example `ssh-keyscan github.com | ssh-keygen -l -f -` produces 43 characters long Base64 outputs (without trailing `=`):
```
# github.com:22 SSH-2.0-babeld-51390fc7
# github.com:22 SSH-2.0-babeld-51390fc7
# github.com:22 SSH-2.0-babeld-51390fc7
# github.com:22 SSH-2.0-babeld-51390fc7
# github.com:22 SSH-2.0-babeld-51390fc7
3072 SHA256:uNiVztksCsDhcc0u9e8BujQXVUpKZIDTMczCvj3tD2s github.com (RSA)
256 SHA256:p2QAMXNIC1TJYWeIOttrVc98/R1BUFWu3/LiyKgUfQM github.com (ECDSA)
256 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU github.com (ED25519)
```


PuTTY does this as well (displays 43 characters long Base64 output without trailing `=`) - when connecting to github.com this text is shown:
```
The host key is not cached for this server:

github.com (port 22)

You have no guarantee that the server is the computer you think it is.

The server's ssh-ed25519 key fingerprint is:

ssh-ed25519 255 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU

If you trust this host, press "Accept" to add the key to PuTTY's cache and carry on connecting.

If you want to carry on connecting just once, without adding the key to the cache, press "Connect Once".

If you do not trust this host, press "Cancel" to abandon the connection.
```


WinSCP also does this - when connecting to github.com it displays these fingerprints:
```
ssh-ed25519 255 +DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU
ssh-ed25519 255 65:96:2d:fc:e8:d5:a9:11:64:0c:0f:ea:00:6e:5b:bd
```

**Review by WojciechNagorski — APPROVED** [review]
LGTM

**Comment by DavisNT** [comment]
@drieseng Can you please review/merge this?

**Comment by DavisNT** [comment]
@drieseng, @WojciechNagorski is there anything I can do to help get this PR approved/merged?

**Comment by WojciechNagorski** [comment]
@DavisNT maybe in the near future I can help.

**Comment by DavisNT** [comment]
@WojciechNagorski, @drieseng Is there anything I can help with to get this PR reviewed and merged?

**Comment by WojciechNagorski** [comment]
I will check it this week.

**Comment by WojciechNagorski** [comment]
Version 2023.0.0 has been published https://www.nuget.org/packages/SSH.NET/2023.0.0

**Comment by WojciechNagorski** [comment]
Version 2023.0.0 has been published https://www.nuget.org/packages/SSH.NET/2023.0.0

**Comment by DavisNT** [comment]
@WojciechNagorski Thank you very much!

**Pull request merged**